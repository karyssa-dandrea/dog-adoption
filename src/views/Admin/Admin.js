import React from 'react';
import { useState } from 'react';
import { insertDog } from '../../services/dogs';
import DogForm from '../../components/Form/Dogform';
import Header from '../../components/Header/Header';

export default function Admin() {
  const [dog, setDog] = useState({});
  const [message, setMessage] = useState('');

  const buttonHandler = async () => {
    await insertDog(dog);
    setDog({});
    setMessage('Success!');
  };

  const updateDogState = (key, value) => {
    dog[key] = value;
    setDog({ ...dog });
  };

  return (
    <div className="admin">
      <Header />
      <DogForm
        buttonHandler={buttonHandler}
        {...dog}
        updateDogState={updateDogState}
        message={message}
      />
    </div>
  );
}
