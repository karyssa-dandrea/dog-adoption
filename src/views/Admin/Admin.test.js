import { render, screen } from '@testing-library/react';
import Admin from '../../views/Admin/Admin';
import { MemoryRouter } from 'react-router-dom';

test('should render admin page', async () => {
  const { container } = render(
    <MemoryRouter>
      <Admin />
    </MemoryRouter>
  );
  await screen.findByText('Add Dog Details');
  expect(container).toMatchSnapshot();
});
