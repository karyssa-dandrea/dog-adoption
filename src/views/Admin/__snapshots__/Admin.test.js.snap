// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render admin page 1`] = `
<div>
  <div
    class="admin"
  >
    <div
      class="header"
    >
      <header>
        <a
          href="/dogs"
        >
           Home 
        </a>
        <a
          href="/admin"
        >
           Admin
        </a>
      </header>
    </div>
    <div
      class="form"
    >
      <h2>
        Add Dog Details
      </h2>
      <p />
      <div
        class="dog-form"
      >
        <label>
          Name
        </label>
        <input
          placeholder="name"
          type="text"
          value=""
        />
        <label>
          Bio:
        </label>
        <input
          placeholder="bio"
          type="text"
          value=""
        />
        <label>
          Age:
        </label>
        <input
          placeholder="age"
          type="text"
          value=""
        />
        <label>
          Image:
        </label>
        <input
          placeholder="image"
          type="text"
          value=""
        />
        <label>
          Breed:
        </label>
        <input
          placeholder="breed"
          type="text"
          value=""
        />
      </div>
      <div
        class="button"
      >
        <button>
          Save
        </button>
      </div>
    </div>
  </div>
</div>
`;
